src/components/Collection.js

```diff
@@ -0,0 +1,227 @@
+import { useState, useEffect } from "react";
+import { useParams } from 'react-router-dom';
+import Book from "./Book";
+import DisplayError from "./DisplayError";
+import LimitMessage from "./LimitMessage";
+import Navigation from "./Navigation";
+import Loader from "./Loader";
+import Pagination from "./Pagination";
+
+ 
+const Collection = ({bookArray, setBookArray}) => {
+
+    const { category } = useParams();
+
+    // Arrays for data storing
+    const [collectionData, setCollectionData] = useState([]);
+    // console.log(collectionData);
+
+    // States for error handling, etc
+    const [isLoading, setIsLoading] = useState(true);
+    const [isError, setIsError] = useState(false);
+    const [overMaxNum, setOverMaxNum] = useState(false);
+
+    // Resource used for Pagination 🦋 "https://medium.com/geekculture/building-a-javascript-pagination-as-simple-as-possible-a9c32dbf4ac1" 🦋
+
+    // State for pagination
+    const [currentPage, setCurrentPage] = useState(1);
+    const [currentPageList, setCurrentPageList] = useState([]);
+    
+    // Pagination
+    const numberOfBooks = collectionData.length;
+    const booksPerPage = 12;
+    const numberOfPages = Math.ceil(numberOfBooks/booksPerPage);
+
+    const paginateCollection = (page) =>{
+
+        setCurrentPage(page);
+        
+        // Define indexes used for slice method used create a shallow copy of the arrays used for pagination
+        const startIndex = (page - 1) * booksPerPage; 
+        const endIndex = startIndex + booksPerPage;
+        
+        // console.log(startIndex, endIndex);
+        setCurrentPageList(collectionData.slice(startIndex,endIndex));
+    }
+
+    const updatePageNum = (e) =>{
+
+        if(e.target.innerHTML === "Next" && currentPage < numberOfPages){
+            setCurrentPage(prev => prev + 1);
+        }else if (e.target.innerHTML === "Prev" && currentPage > 1){
+            setCurrentPage(prev => prev -1);
+        }
+
+    }
+
+    useEffect(()=>{
+        paginateCollection(currentPage);
+    },[currentPage, collectionData]); // eslint-disable-line react-hooks/exhaustive-deps
+
+
+    // Search parameters
+    const url = new URL("https://openlibrary.org/search.json");
+    url.search = new URLSearchParams({
+        subject: "picture books",
+        q: category,
+        audience: "juvenile"
+    });
+
+    useEffect(() => {
+
+        const getBookData = async () => {
+
+            setIsError(false); 
+
+            try {
+                //Initial call is made here:
```

Comment:
I appreciate this pseudocode and the bits below!  Helping me to better understand what I'm seeing.